### PR TITLE
Monitors

### PR DESCRIPTION
**Pull request by MuhammadKhalilzadeh**
Monitors: hosts, with some bugs and challenges to take care of that I'll try to work it out as my next step

screencapture-localhost-5173-monitors-2024-06-13-10_34_28

### CODE REVIEW

**Review by ajhollid — CHANGES_REQUESTED** [review]
Overall this looks good.  I think we can leave the HTML and CSS charts for now, and we can work on integrating ChartJS in the future.

The monitor page will make an API request to the backend and receive an `Array<Monitor>` in response.

We need the `CurrentMonitors` component to take that `Array<Monitor>` as a prop so that we can populate the `HostTable` component with our `ArrayList<Monitor>` instead of hardcoded data.

In summary, we need an `ArrayList<Monitor>` to be passed as props from `MonitorPage` to `HostTable`

`MonitorPage` -> `CurrentMonitors` -> `HostTable`

and then have `HostTable` render those monitors dynamically

**Comment by MuhammadKhalilzadeh** [comment]
June 19th
This commit and review request is still not for the merge.
I paused the progress in the `Create New Monitor` page and switch back to the `Host table`.
I made the changes in the UI according to what Gorkem said for some `gaps`, `strokes` and `sizings`.
The `Dashboard setting menu` for `Actions` is added and styles are changed accordingly.
For fetching the data from API and casting it to display in the Hosts table, I created a `mock_data.json` file and generated a mock data according to the sample data from `**Alex**`. Please make sure I'm casting it correctly inside `CurrentStats.jsx` file, since this is the main purpose of this commit.

**Comment by ajhollid** [comment]
This looks good, I believe you are on the correct track here.

Next you can pass that data to the `BarChart` component and get it displaying the response time :+1:

**Review by ajhollid — COMMENTED** [review]
Looks good, this is on the right track

File: Client/src/Pages/Monitors/CurrentStats.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    ),
+    status: HostStatus(
+      "var(--env-var-color-15)",
+      "Cannot resolve",
+      "var(--env-var-color-22)"
+    ),
+    team: <BarChart />,
+    actions: HostActions(),
+  },
+];
```
Comment (ajhollid) [inline Client/src/Pages/Monitors/CurrentStats.jsx:87]: I don't think we need this hardcoded data anymore since we have at least mock data now right?

**Review by ajhollid — APPROVED** [review]
I think this looks good, we can probably remove the hardcoded data.

As for the merge conflicts, I can resolve them when we are ready to merge.

File: Client/src/Pages/Monitors/CurrentStats.jsx (hunk 1 above)
Comment (MuhammadKhalilzadeh) [reply Client/src/Pages/Monitors/CurrentStats.jsx:87]: Yes. I'll remove it

Comment (MuhammadKhalilzadeh) [reply Client/src/Pages/Monitors/CurrentStats.jsx]: Done

**Review by ajhollid — APPROVED** [review]
Pending reviews from the front end team I think this is ready to merge

**Review by danielcj2 — APPROVED** [review]
Looks good to me

**Comment by MuhammadKhalilzadeh** [comment]
@Skorpios604 What do you think?
